hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -255,7 +255,8 @@ def test_function(self, data):
 
         if (
             self.finish_shrinking_deadline is not None
-            and self.finish_shrinking_deadline < perf_counter()
+            and self.finish_shrinking_deadline < time.perf_counter()
+            and False
```

Comment:
This bit seems leftover?

Reply:
Doh. And the fix doesn't seem to work once I remove that. I may have conned myself here. More investigation required.

Reply:
(Or rather I do know that time mocking wasn't working because I saw the test fail much more often when debugging was turned on, but there may be more going on)

Reply:
Ugh, right. OK. I understand now. There are two problems.

1. The time mocking genuinely wasn't working and that's why this was flaky and performance dependent.
2. ALSO once you fix the time mocking this test fails because it's too slow.

So what I'm going to do is put the skip back in, merge this as is, and then address (2) in a second pull request.

Reply:
Sounds good :+1: